### PR TITLE
Fix otsu

### PR DESCRIPTION
**Pull request by TitouanGragnic**
<!> not in dev

Compute threashold in CPU is ok and don't lose FPS

### CODE REVIEW

File: Holovibes/sources/analysis/analysis.cc
Comment (ArthurCourselle) [inline Holovibes/sources/analysis/analysis.cc:1]: Please remove all the dead code ... ;(

File: Holovibes/cuda_sources/analysis/otsu.cu [hunk 1]
```diff
@@ -19,13 +19,16 @@ __global__ void histogram_kernel(const float* image, uint* hist, int imgSize)
...
     __syncthreads();
 
     // Populate shared histogram
     if (idx < imgSize)
     {
-        int bin = static_cast<int>(image[idx] * NUM_BINS);
-        atomicAdd(&shared_hist[bin], 1);
+        int bin = static_cast<int>(image[idx] * (NUM_BINS - 1));
+        atomicAdd(shared_hist + bin, 1);
+        atomicAdd(shared_hist + NUM_BINS, 1);
```
Comment (ArthurCourselle) [inline Holovibes/cuda_sources/analysis/otsu.cu]: could you use [] operator instead of +

**Review by ArthurCourselle — CHANGES_REQUESTED** [review]

**Review by ArthurCourselle — APPROVED** [review]

File: Holovibes/cuda_sources/analysis/otsu.cu [hunk 2]
```diff
@@ -79,85 +81,58 @@ __global__ void bradley_threshold_kernel(float* output,
...
+    cudaXMemcpy(histogram, histo_buffer_d, NUM_BINS * sizeof(uint), cudaMemcpyDeviceToHost);
+
+    std::vector<double> probabilities(256, 0.0);
+    for (int i = 0; i < 256; ++i)
     {
-        if (tid < offset)
-            partial_sum[tid] += partial_sum[tid + offset];
-        __syncthreads();
+        probabilities[i] = static_cast<double>(histogram[i]) / size;
     }
```
Comment (Baschtien) [inline Holovibes/cuda_sources/analysis/otsu.cu:101]: The brackets {} are not needed for the loop anymore.

File: Holovibes/sources/analysis/analysis.cc
Comment (Baschtien) [inline Holovibes/sources/analysis/analysis.cc:1]: Maybe don't comment all the csv code ? It may be easier for the Analysis team to merge that way, idk...

Comment (TitouanGragnic) [reply Holovibes/sources/analysis/analysis.cc:1]: They comment this part of the code for me, they are going to do the merge don't worry

Comment (Baschtien) [reply Holovibes/sources/analysis/analysis.cc:1]: Ok!

**Review by GustaveHerve — APPROVED** [review]

**Pull request merged**